src/main/java/seedu/multitasky/logic/CommandHistory.java

```diff
@@ -30,4 +39,18 @@ public void add(String userInput) {
         return new ArrayList<>(userInputHistory);
     }
 
+    // @@author A0140633R
+    public Set<String> getPrevSearch() {
+        return previousSearch;
+    }
+
+    public State getPrevState() {
+        return previousState;
+    }
+
+    public void setNextSearch(Set<String> nextSearch, State nextState) {
+        previousSearch = nextSearch;
+        previousState = nextState;
+    }
```

Comment:
haha , thats actually what i named it as initially. i forgot why i changed it to next search